// @lshay/ui/dist/cjs/components/new-york/ui/breadcrumb.js
'use strict';

var e = require('react');
var reactIcons = require('@radix-ui/react-icons');
var reactSlot = require('@radix-ui/react-slot');
var clsx = require('clsx');
var tailwindMerge = require('tailwind-merge');

function _interopNamespace(e) {
	if (e && e.__esModule) return e;
	var n = Object.create(null);
	if (e) {
		Object.keys(e).forEach(function (k) {
			if (k !== 'default') {
				var d = Object.getOwnPropertyDescriptor(e, k);
				Object.defineProperty(n, k, d.get ? d : {
					enumerable: true,
					get: function () { return e[k]; }
				});
			}
		});
	}
	n.default = e;
	return Object.freeze(n);
}

var e__namespace = /*#__PURE__*/_interopNamespace(e);

function o(...a){return tailwindMerge.twMerge(clsx.clsx(a))}var p=e__namespace.forwardRef(({...a},r)=>e__namespace.createElement("nav",{ref:r,"aria-label":"breadcrumb",...a}));p.displayName="Breadcrumb";var d=e__namespace.forwardRef(({className:a,...r},t)=>e__namespace.createElement("ol",{ref:t,className:o("flex flex-wrap items-center gap-1.5 break-words text-sm text-muted-foreground sm:gap-2.5",a),...r}));d.displayName="BreadcrumbList";var f=e__namespace.forwardRef(({className:a,...r},t)=>e__namespace.createElement("li",{ref:t,className:o("inline-flex items-center gap-1.5",a),...r}));f.displayName="BreadcrumbItem";var u=e__namespace.forwardRef(({asChild:a,className:r,...t},s)=>e__namespace.createElement(a?reactSlot.Slot:"a",{ref:s,className:o("transition-colors hover:text-foreground",r),...t}));u.displayName="BreadcrumbLink";var R=e__namespace.forwardRef(({className:a,...r},t)=>e__namespace.createElement("span",{ref:t,role:"link","aria-disabled":"true","aria-current":"page",className:o("font-normal text-foreground",a),...r}));R.displayName="BreadcrumbPage";var b=({children:a,className:r,...t})=>e__namespace.createElement("li",{role:"presentation","aria-hidden":"true",className:o("[&>svg]:size-3.5",r),...t},a??e__namespace.createElement(reactIcons.ChevronRightIcon,null));b.displayName="BreadcrumbSeparator";var N=({className:a,...r})=>e__namespace.createElement("span",{role:"presentation","aria-hidden":"true",className:o("flex h-9 w-9 items-center justify-center",a),...r},e__namespace.createElement(reactIcons.DotsHorizontalIcon,{className:"h-4 w-4"}),e__namespace.createElement("span",{className:"sr-only"},"More"));N.displayName="BreadcrumbElipssis";

exports.Breadcrumb = p;
exports.BreadcrumbEllipsis = N;
exports.BreadcrumbItem = f;
exports.BreadcrumbLink = u;
exports.BreadcrumbList = d;
exports.BreadcrumbPage = R;
exports.BreadcrumbSeparator = b;
