// @lshay/ui/dist/cjs/components/new-york/ui/drawer.js
'use strict';

var e = require('react');
var vaul = require('vaul');
var clsx = require('clsx');
var tailwindMerge = require('tailwind-merge');

function _interopNamespace(e) {
	if (e && e.__esModule) return e;
	var n = Object.create(null);
	if (e) {
		Object.keys(e).forEach(function (k) {
			if (k !== 'default') {
				var d = Object.getOwnPropertyDescriptor(e, k);
				Object.defineProperty(n, k, d.get ? d : {
					enumerable: true,
					get: function () { return e[k]; }
				});
			}
		});
	}
	n.default = e;
	return Object.freeze(n);
}

var e__namespace = /*#__PURE__*/_interopNamespace(e);

function i(...t){return tailwindMerge.twMerge(clsx.clsx(t))}var c=({shouldScaleBackground:t=!0,...r})=>e__namespace.createElement(vaul.Drawer.Root,{shouldScaleBackground:t,...r});c.displayName="Drawer";var g=vaul.Drawer.Trigger,p=vaul.Drawer.Portal,C=vaul.Drawer.Close,s=e__namespace.forwardRef(({className:t,...r},o)=>e__namespace.createElement(vaul.Drawer.Overlay,{ref:o,className:i("fixed inset-0 z-50 bg-black/80",t),...r}));s.displayName=vaul.Drawer.Overlay.displayName;var f=e__namespace.forwardRef(({className:t,children:r,...o},n)=>e__namespace.createElement(p,null,e__namespace.createElement(s,null),e__namespace.createElement(vaul.Drawer.Content,{ref:n,className:i("fixed inset-x-0 bottom-0 z-50 mt-24 flex h-auto flex-col rounded-t-[10px] border bg-background",t),...o},e__namespace.createElement("div",{className:"mx-auto mt-4 h-2 w-[100px] rounded-full bg-muted"}),r)));f.displayName="DrawerContent";var d=({className:t,...r})=>e__namespace.createElement("div",{className:i("grid gap-1.5 p-4 text-center sm:text-left",t),...r});d.displayName="DrawerHeader";var D=({className:t,...r})=>e__namespace.createElement("div",{className:i("mt-auto flex flex-col gap-2 p-4",t),...r});D.displayName="DrawerFooter";var w=e__namespace.forwardRef(({className:t,...r},o)=>e__namespace.createElement(vaul.Drawer.Title,{ref:o,className:i("text-lg font-semibold leading-none tracking-tight",t),...r}));w.displayName=vaul.Drawer.Title.displayName;var R=e__namespace.forwardRef(({className:t,...r},o)=>e__namespace.createElement(vaul.Drawer.Description,{ref:o,className:i("text-sm text-muted-foreground",t),...r}));R.displayName=vaul.Drawer.Description.displayName;

exports.Drawer = c;
exports.DrawerClose = C;
exports.DrawerContent = f;
exports.DrawerDescription = R;
exports.DrawerFooter = D;
exports.DrawerHeader = d;
exports.DrawerOverlay = s;
exports.DrawerPortal = p;
exports.DrawerTitle = w;
exports.DrawerTrigger = g;
