// @lshay/ui/dist/cjs/components/new-york/ui/pagination.js
'use strict';

var e = require('react');
var reactIcons = require('@radix-ui/react-icons');
var clsx = require('clsx');
var tailwindMerge = require('tailwind-merge');
var reactSlot = require('@radix-ui/react-slot');
var classVarianceAuthority = require('class-variance-authority');

function _interopNamespace(e) {
	if (e && e.__esModule) return e;
	var n = Object.create(null);
	if (e) {
		Object.keys(e).forEach(function (k) {
			if (k !== 'default') {
				var d = Object.getOwnPropertyDescriptor(e, k);
				Object.defineProperty(n, k, d.get ? d : {
					enumerable: true,
					get: function () { return e[k]; }
				});
			}
		});
	}
	n.default = e;
	return Object.freeze(n);
}

var e__namespace = /*#__PURE__*/_interopNamespace(e);

function n(...t){return tailwindMerge.twMerge(clsx.clsx(t))}var r=classVarianceAuthority.cva("inline-flex items-center justify-center whitespace-nowrap rounded-md text-sm font-medium transition-colors focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:pointer-events-none disabled:opacity-50",{variants:{variant:{default:"bg-primary text-primary-foreground shadow hover:bg-primary/90",destructive:"bg-destructive text-destructive-foreground shadow-sm hover:bg-destructive/90",outline:"border border-input bg-background shadow-sm hover:bg-accent hover:text-accent-foreground",secondary:"bg-secondary text-secondary-foreground shadow-sm hover:bg-secondary/80",ghost:"hover:bg-accent hover:text-accent-foreground",link:"text-primary underline-offset-4 hover:underline"},size:{default:"h-9 px-4 py-2",sm:"h-8 rounded-md px-3 text-xs",lg:"h-10 rounded-md px-8",icon:"h-9 w-9"}},defaultVariants:{variant:"default",size:"default"}}),g=e__namespace.forwardRef(({className:t,variant:a,size:o,asChild:s=!1,...c},l)=>e__namespace.createElement(s?reactSlot.Slot:"button",{className:n(r({variant:a,size:o,className:t})),ref:l,...c}));g.displayName="Button";var x=({className:t,...a})=>e__namespace.createElement("nav",{role:"navigation","aria-label":"pagination",className:n("mx-auto flex w-full justify-center",t),...a});x.displayName="Pagination";var y=e__namespace.forwardRef(({className:t,...a},o)=>e__namespace.createElement("ul",{ref:o,className:n("flex flex-row items-center gap-1",t),...a}));y.displayName="PaginationContent";var b=e__namespace.forwardRef(({className:t,...a},o)=>e__namespace.createElement("li",{ref:o,className:n("",t),...a}));b.displayName="PaginationItem";var p=({className:t,isActive:a,size:o="icon",...s})=>e__namespace.createElement("a",{"aria-current":a?"page":void 0,className:n(r({variant:a?"outline":"ghost",size:o}),t),...s});p.displayName="PaginationLink";var N=({className:t,...a})=>e__namespace.createElement(p,{"aria-label":"Go to previous page",size:"default",className:n("gap-1 pl-2.5",t),...a},e__namespace.createElement(reactIcons.ChevronLeftIcon,{className:"h-4 w-4"}),e__namespace.createElement("span",null,"Previous"));N.displayName="PaginationPrevious";var w=({className:t,...a})=>e__namespace.createElement(p,{"aria-label":"Go to next page",size:"default",className:n("gap-1 pr-2.5",t),...a},e__namespace.createElement("span",null,"Next"),e__namespace.createElement(reactIcons.ChevronRightIcon,{className:"h-4 w-4"}));w.displayName="PaginationNext";var C=({className:t,...a})=>e__namespace.createElement("span",{"aria-hidden":!0,className:n("flex h-9 w-9 items-center justify-center",t),...a},e__namespace.createElement(reactIcons.DotsHorizontalIcon,{className:"h-4 w-4"}),e__namespace.createElement("span",{className:"sr-only"},"More pages"));C.displayName="PaginationEllipsis";

exports.Pagination = x;
exports.PaginationContent = y;
exports.PaginationEllipsis = C;
exports.PaginationItem = b;
exports.PaginationLink = p;
exports.PaginationNext = w;
exports.PaginationPrevious = N;
