// @lshay/ui/dist/cjs/components/new-york/ui/sheet.js
'use strict';

var t = require('react');
var e = require('@radix-ui/react-dialog');
var reactIcons = require('@radix-ui/react-icons');
var classVarianceAuthority = require('class-variance-authority');
var clsx = require('clsx');
var tailwindMerge = require('tailwind-merge');

function _interopNamespace(e) {
	if (e && e.__esModule) return e;
	var n = Object.create(null);
	if (e) {
		Object.keys(e).forEach(function (k) {
			if (k !== 'default') {
				var d = Object.getOwnPropertyDescriptor(e, k);
				Object.defineProperty(n, k, d.get ? d : {
					enumerable: true,
					get: function () { return e[k]; }
				});
			}
		});
	}
	n.default = e;
	return Object.freeze(n);
}

var t__namespace = /*#__PURE__*/_interopNamespace(t);
var e__namespace = /*#__PURE__*/_interopNamespace(e);

function s(...i){return tailwindMerge.twMerge(clsx.clsx(i))}var w=e__namespace.Root,D=e__namespace.Trigger,E=e__namespace.Close,f=e__namespace.Portal,r=t__namespace.forwardRef(({className:i,...o},a)=>t__namespace.createElement(e__namespace.Overlay,{className:s("fixed inset-0 z-50 bg-black/80  data-[state=open]:animate-in data-[state=closed]:animate-out data-[state=closed]:fade-out-0 data-[state=open]:fade-in-0",i),...o,ref:a}));r.displayName=e__namespace.Overlay.displayName;var h=classVarianceAuthority.cva("fixed z-50 gap-4 bg-background p-6 shadow-lg transition ease-in-out data-[state=open]:animate-in data-[state=closed]:animate-out data-[state=closed]:duration-300 data-[state=open]:duration-500",{variants:{side:{top:"inset-x-0 top-0 border-b data-[state=closed]:slide-out-to-top data-[state=open]:slide-in-from-top",bottom:"inset-x-0 bottom-0 border-t data-[state=closed]:slide-out-to-bottom data-[state=open]:slide-in-from-bottom",left:"inset-y-0 left-0 h-full w-3/4 border-r data-[state=closed]:slide-out-to-left data-[state=open]:slide-in-from-left sm:max-w-sm",right:"inset-y-0 right-0 h-full w-3/4 border-l data-[state=closed]:slide-out-to-right data-[state=open]:slide-in-from-right sm:max-w-sm"}},defaultVariants:{side:"right"}}),u=t__namespace.forwardRef(({side:i="right",className:o,children:a,...n},m)=>t__namespace.createElement(f,null,t__namespace.createElement(r,null),t__namespace.createElement(e__namespace.Content,{ref:m,className:s(h({side:i}),o),...n},a,t__namespace.createElement(e__namespace.Close,{className:"absolute right-4 top-4 rounded-sm opacity-70 ring-offset-background transition-opacity hover:opacity-100 focus:outline-none focus:ring-2 focus:ring-ring focus:ring-offset-2 disabled:pointer-events-none data-[state=open]:bg-secondary"},t__namespace.createElement(reactIcons.Cross2Icon,{className:"h-4 w-4"}),t__namespace.createElement("span",{className:"sr-only"},"Close")))));u.displayName=e__namespace.Content.displayName;var S=({className:i,...o})=>t__namespace.createElement("div",{className:s("flex flex-col space-y-2 text-center sm:text-left",i),...o});S.displayName="SheetHeader";var v=({className:i,...o})=>t__namespace.createElement("div",{className:s("flex flex-col-reverse sm:flex-row sm:justify-end sm:space-x-2",i),...o});v.displayName="SheetFooter";var y=t__namespace.forwardRef(({className:i,...o},a)=>t__namespace.createElement(e__namespace.Title,{ref:a,className:s("text-lg font-semibold text-foreground",i),...o}));y.displayName=e__namespace.Title.displayName;var P=t__namespace.forwardRef(({className:i,...o},a)=>t__namespace.createElement(e__namespace.Description,{ref:a,className:s("text-sm text-muted-foreground",i),...o}));P.displayName=e__namespace.Description.displayName;

exports.Sheet = w;
exports.SheetClose = E;
exports.SheetContent = u;
exports.SheetDescription = P;
exports.SheetFooter = v;
exports.SheetHeader = S;
exports.SheetOverlay = r;
exports.SheetPortal = f;
exports.SheetTitle = y;
exports.SheetTrigger = D;
