// @lshay/ui/dist/cjs/components/default/ui/card.js
'use strict';

var e = require('react');
var clsx = require('clsx');
var tailwindMerge = require('tailwind-merge');

function _interopNamespace(e) {
	if (e && e.__esModule) return e;
	var n = Object.create(null);
	if (e) {
		Object.keys(e).forEach(function (k) {
			if (k !== 'default') {
				var d = Object.getOwnPropertyDescriptor(e, k);
				Object.defineProperty(n, k, d.get ? d : {
					enumerable: true,
					get: function () { return e[k]; }
				});
			}
		});
	}
	n.default = e;
	return Object.freeze(n);
}

var e__namespace = /*#__PURE__*/_interopNamespace(e);

function s(...t){return tailwindMerge.twMerge(clsx.clsx(t))}var n=e__namespace.forwardRef(({className:t,...a},r)=>e__namespace.createElement("div",{ref:r,className:s("rounded-lg border bg-card text-card-foreground shadow-sm",t),...a}));n.displayName="Card";var i=e__namespace.forwardRef(({className:t,...a},r)=>e__namespace.createElement("div",{ref:r,className:s("flex flex-col space-y-1.5 p-6",t),...a}));i.displayName="CardHeader";var c=e__namespace.forwardRef(({className:t,...a},r)=>e__namespace.createElement("h3",{ref:r,className:s("text-2xl font-semibold leading-none tracking-tight",t),...a}));c.displayName="CardTitle";var l=e__namespace.forwardRef(({className:t,...a},r)=>e__namespace.createElement("p",{ref:r,className:s("text-sm text-muted-foreground",t),...a}));l.displayName="CardDescription";var m=e__namespace.forwardRef(({className:t,...a},r)=>e__namespace.createElement("div",{ref:r,className:s("p-6 pt-0",t),...a}));m.displayName="CardContent";var f=e__namespace.forwardRef(({className:t,...a},r)=>e__namespace.createElement("div",{ref:r,className:s("flex items-center p-6 pt-0",t),...a}));f.displayName="CardFooter";

exports.Card = n;
exports.CardContent = m;
exports.CardDescription = l;
exports.CardFooter = f;
exports.CardHeader = i;
exports.CardTitle = c;
