// @lshay/ui/dist/cjs/components/default/ui/pagination.js
'use strict';

var t = require('react');
var lucideReact = require('lucide-react');
var clsx = require('clsx');
var tailwindMerge = require('tailwind-merge');
var reactSlot = require('@radix-ui/react-slot');
var classVarianceAuthority = require('class-variance-authority');

function _interopNamespace(e) {
	if (e && e.__esModule) return e;
	var n = Object.create(null);
	if (e) {
		Object.keys(e).forEach(function (k) {
			if (k !== 'default') {
				var d = Object.getOwnPropertyDescriptor(e, k);
				Object.defineProperty(n, k, d.get ? d : {
					enumerable: true,
					get: function () { return e[k]; }
				});
			}
		});
	}
	n.default = e;
	return Object.freeze(n);
}

var t__namespace = /*#__PURE__*/_interopNamespace(t);

function n(...e){return tailwindMerge.twMerge(clsx.clsx(e))}var s=classVarianceAuthority.cva("inline-flex items-center justify-center whitespace-nowrap rounded-md text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50",{variants:{variant:{default:"bg-primary text-primary-foreground hover:bg-primary/90",destructive:"bg-destructive text-destructive-foreground hover:bg-destructive/90",outline:"border border-input bg-background hover:bg-accent hover:text-accent-foreground",secondary:"bg-secondary text-secondary-foreground hover:bg-secondary/80",ghost:"hover:bg-accent hover:text-accent-foreground",link:"text-primary underline-offset-4 hover:underline"},size:{default:"h-10 px-4 py-2",sm:"h-9 rounded-md px-3",lg:"h-11 rounded-md px-8",icon:"h-10 w-10"}},defaultVariants:{variant:"default",size:"default"}}),g=t__namespace.forwardRef(({className:e,variant:a,size:o,asChild:r=!1,...l},c)=>t__namespace.createElement(r?reactSlot.Slot:"button",{className:n(s({variant:a,size:o,className:e})),ref:c,...l}));g.displayName="Button";var y=({className:e,...a})=>t__namespace.createElement("nav",{role:"navigation","aria-label":"pagination",className:n("mx-auto flex w-full justify-center",e),...a});y.displayName="Pagination";var x=t__namespace.forwardRef(({className:e,...a},o)=>t__namespace.createElement("ul",{ref:o,className:n("flex flex-row items-center gap-1",e),...a}));x.displayName="PaginationContent";var h=t__namespace.forwardRef(({className:e,...a},o)=>t__namespace.createElement("li",{ref:o,className:n("",e),...a}));h.displayName="PaginationItem";var p=({className:e,isActive:a,size:o="icon",...r})=>t__namespace.createElement("a",{"aria-current":a?"page":void 0,className:n(s({variant:a?"outline":"ghost",size:o}),e),...r});p.displayName="PaginationLink";var N=({className:e,...a})=>t__namespace.createElement(p,{"aria-label":"Go to previous page",size:"default",className:n("gap-1 pl-2.5",e),...a},t__namespace.createElement(lucideReact.ChevronLeft,{className:"h-4 w-4"}),t__namespace.createElement("span",null,"Previous"));N.displayName="PaginationPrevious";var C=({className:e,...a})=>t__namespace.createElement(p,{"aria-label":"Go to next page",size:"default",className:n("gap-1 pr-2.5",e),...a},t__namespace.createElement("span",null,"Next"),t__namespace.createElement(lucideReact.ChevronRight,{className:"h-4 w-4"}));C.displayName="PaginationNext";var R=({className:e,...a})=>t__namespace.createElement("span",{"aria-hidden":!0,className:n("flex h-9 w-9 items-center justify-center",e),...a},t__namespace.createElement(lucideReact.MoreHorizontal,{className:"h-4 w-4"}),t__namespace.createElement("span",{className:"sr-only"},"More pages"));R.displayName="PaginationEllipsis";

exports.Pagination = y;
exports.PaginationContent = x;
exports.PaginationEllipsis = R;
exports.PaginationItem = h;
exports.PaginationLink = p;
exports.PaginationNext = C;
exports.PaginationPrevious = N;
