// @lshay/ui/dist/cjs/components/default/ui/table.js
'use strict';

var e = require('react');
var clsx = require('clsx');
var tailwindMerge = require('tailwind-merge');

function _interopNamespace(e) {
	if (e && e.__esModule) return e;
	var n = Object.create(null);
	if (e) {
		Object.keys(e).forEach(function (k) {
			if (k !== 'default') {
				var d = Object.getOwnPropertyDescriptor(e, k);
				Object.defineProperty(n, k, d.get ? d : {
					enumerable: true,
					get: function () { return e[k]; }
				});
			}
		});
	}
	n.default = e;
	return Object.freeze(n);
}

var e__namespace = /*#__PURE__*/_interopNamespace(e);

function o(...t){return tailwindMerge.twMerge(clsx.clsx(t))}var T=e__namespace.forwardRef(({className:t,...a},l)=>e__namespace.createElement("div",{className:"relative w-full overflow-auto"},e__namespace.createElement("table",{ref:l,className:o("w-full caption-bottom text-sm",t),...a})));T.displayName="Table";var m=e__namespace.forwardRef(({className:t,...a},l)=>e__namespace.createElement("thead",{ref:l,className:o("[&_tr]:border-b",t),...a}));m.displayName="TableHeader";var c=e__namespace.forwardRef(({className:t,...a},l)=>e__namespace.createElement("tbody",{ref:l,className:o("[&_tr:last-child]:border-0",t),...a}));c.displayName="TableBody";var b=e__namespace.forwardRef(({className:t,...a},l)=>e__namespace.createElement("tfoot",{ref:l,className:o("border-t bg-muted/50 font-medium [&>tr]:last:border-b-0",t),...a}));b.displayName="TableFooter";var n=e__namespace.forwardRef(({className:t,...a},l)=>e__namespace.createElement("tr",{ref:l,className:o("border-b transition-colors hover:bg-muted/50 data-[state=selected]:bg-muted",t),...a}));n.displayName="TableRow";var d=e__namespace.forwardRef(({className:t,...a},l)=>e__namespace.createElement("th",{ref:l,className:o("h-12 px-4 text-left align-middle font-medium text-muted-foreground [&:has([role=checkbox])]:pr-0",t),...a}));d.displayName="TableHead";var i=e__namespace.forwardRef(({className:t,...a},l)=>e__namespace.createElement("td",{ref:l,className:o("p-4 align-middle [&:has([role=checkbox])]:pr-0",t),...a}));i.displayName="TableCell";var f=e__namespace.forwardRef(({className:t,...a},l)=>e__namespace.createElement("caption",{ref:l,className:o("mt-4 text-sm text-muted-foreground",t),...a}));f.displayName="TableCaption";

exports.Table = T;
exports.TableBody = c;
exports.TableCaption = f;
exports.TableCell = i;
exports.TableFooter = b;
exports.TableHead = d;
exports.TableHeader = m;
exports.TableRow = n;
